service/controller/key/common.go

```diff
@@ -154,11 +154,17 @@ func RegionARN(region string) string {
 }
 
 func RoleARNMaster(getter LabelsGetter, region string, accountID string) string {
-	return baseRoleARN(getter, region, accountID, "master")
+	clusterID := ClusterID(getter)
+	partition := RegionARN(region)
+
+	return fmt.Sprintf("arn:%s:iam::%s:role/%s-master-%s", partition, accountID, clusterID, EC2RoleK8s)
 }
 
 func RoleARNWorker(getter LabelsGetter, region string, accountID string) string {
-	return baseRoleARN(getter, region, accountID, "worker")
+	clusterID := ClusterID(getter)
+	partition := RegionARN(region)
+
+	return fmt.Sprintf("arn:%s:iam::%s:role/gs-cluster-%s-role-*", partition, accountID, clusterID)
```

Comment:
How are these roles actually created? How to verify these are the right ones?

Reply:
Roles are created via CF. This is a format used in CF

Reply:
Yes since the role names are defined inline in the CF template this makes sense to me

Maybe it makes sense to move them to a key func and pass the whole role name to the template. But I think that can be done as a later improvement.

Reply:
But that's already in key func. I've just changed existing key func for role names

Reply:
Wow yes somehow I missed this is the key func. 🤦‍♂ 

OK all good thanks.